spec/haskell/src/SEL4/Object/Reply.lhs

```diff
@@ -80,14 +80,14 @@ This module specifies the behavior of reply objects.
 >     when (nextReplyPtrOpt /= Nothing) $ do
 >         assert (isHead nextReplyPtrOpt) "the reply must be at the head"
 >         scPtr <- return $ theHeadScPtr nextReplyPtrOpt
->         tcbScPtrOpt <- threadGet tcbSchedContext tcbPtr
->         when (tcbScPtrOpt == Nothing) $ schedContextDonate scPtr tcbPtr
 >         sc <- getSchedContext scPtr
 >         setSchedContext scPtr (sc { scReply = prevReplyPtrOpt })
 >         when (prevReplyPtrOpt /= Nothing) $ do
 >             prevReplyPtr <- return $ fromJust prevReplyPtrOpt
->             prevReply <- getReply prevReplyPtr
->             setReply prevReplyPtr (prevReply { replyNext = replyNext reply })
+>             updateReply prevReplyPtr (\replyPrev -> replyPrev { replyNext = replyNext reply })
```

Comment:
tiniest of nitpicks: I like to keep them consistent, so I would prefer `prevReply` to match `prevReplyPtr` (or `replyPrev` and `replyPrevPtr`)